src/main/java/com/iota/iri/service/milestone/MilestoneSolidifier.java

```diff
@@ -0,0 +1,330 @@
+     * milestone index).
+     */
+    private void determineYoungestMilestoneInQueue() {
+        youngestMilestoneInQueue = null;
+        for (Map.Entry<Hash, Integer> currentEntry : milestonesToSolidify.entrySet()) {
+            if (youngestMilestoneInQueue == null || currentEntry.getValue() > youngestMilestoneInQueue.getValue()) {
+                youngestMilestoneInQueue = currentEntry;
+            }
+        }
+    }
+
+    /**
+     * This method returns the earliest seen Milestone from the unsolid milestones pool, that is not part of the
+     * {@link #milestonesToSolidify} queue yet.
+     *
+     * It simply iterates over all milestones in the pool and looks for the one with the lowest index, that is not
+     * getting actively solidified, yet.
+     *
+     * @return the Map.Entry holding the earliest milestone or null if the pool does not contain any new candidates.
+     */
+    private Map.Entry<Hash, Integer> getNextSolidificationCandidate() {
+        Map.Entry<Hash, Integer> nextSolidificationCandidate = null;
+        for (Map.Entry<Hash, Integer> milestoneEntry : unsolidMilestonesPool.entrySet()) {
+            if (!milestonesToSolidify.containsKey(milestoneEntry.getKey()) && (nextSolidificationCandidate == null ||
+                    milestoneEntry.getValue() < nextSolidificationCandidate.getValue())) {
+
+                nextSolidificationCandidate = milestoneEntry;
+            }
+        }
+
+        return nextSolidificationCandidate;
+    }
+
+    /**
+     * This method performs the actual solidity check on the selected milestone.
+     *
+     * It first dumps a log message to keep the node operator informed about the progress of solidification, and then
+     * issues the {@link TransactionValidator#checkSolidity(Hash, boolean, int)} call that starts the solidification
+     * process.
+     *
+     * We limit the amount of transactions that may be processed during the solidity check, since we want to solidify
+     * from the oldest milestone to the newest one and not "block" the solidification with a very recent milestone that
+     * needs to traverse huge chunks of the tangle. The main goal of this is to give the solidification just enough
+     * "resources" to discover the previous milestone while at the same time allowing fast solidity checks.
+     *
+     * @param currentEntry milestone entry that shall be checked
+     * @return true if the given milestone is solid or false otherwise
+     */
+    private boolean isSolid(Map.Entry<Hash, Integer> currentEntry) {
+        if (unsolidMilestonesPool.size() > 1) {
+            statusLogger.info("Solidifying milestone #" + currentEntry.getValue() +
+                    " [" + milestonesToSolidify.size() + " / " + unsolidMilestonesPool.size() + "]");
+        }
+
+        try {
+            return transactionValidator.checkSolidity(currentEntry.getKey(), true,
+                    SOLIDIFICATION_TRANSACTIONS_LIMIT * 4);
```

Comment:
huh why `* 4`?

Reply:
LOL that was a bug - looks like i missed to replace the file in this PR with the updated one from the dev branch.

Its fixed (default level slighly increased an no * 4)